packages/@aws-cdk/custom-resources/lib/provider-framework/runtime/util.ts

```diff
@@ -0,0 +1,36 @@
+// tslint:disable: no-console
+import { CloudFormationEventContext, submitCloudFormationResponse } from './cfn-response';
+
+export let includeStackTraces = true; // for unit tests
+
+export async function failOnError(event: CloudFormationEventContext, block: () => Promise<any>) {
+  try {
+    return await block();
+  } catch (e) {
+    // tell waiter state machine to retry
+    if (e instanceof Retry) { throw e; }
+
+    // this is an actual error, fail the activity altogether and exist.
+    await submitCloudFormationResponse('FAILED', event, {
+      reason: includeStackTraces ? e.stack : e.message,
+    });
+
+    return {
+      error: e.stack
```

Comment:
Is someone going to look at this return value? Why does this not depend on `includeStackTraces`?

Reply:
Not really. I'll just return